zipper/protocols/victoriametrics/fetch.go

```diff
@@ -25,7 +25,13 @@ func (c *VictoriaMetricsGroup) Fetch(ctx context.Context, request *protov3.Multi
 	}
 	logger := c.logger.With(zap.String("type", "fetch"), zap.String("request", request.String()))
 	stats := &types.Stats{}
-	rewrite, _ := url.Parse("http://127.0.0.1/api/v1/query_range")
+	var serverUrl string
+	if c.vmClusterTenantId >= 0 {
```

Comment:
I think it should be > 0? Or you should put "-1" as default in example config, which should work out of the box for non-clustered version IMO

Reply:
Hi, @deniszh 

Tenant ID 0 is legal id, see https://docs.victoriametrics.com/Cluster-VictoriaMetrics.html#multitenancy :

* Each accountID and projectID is identified by an arbitrary 32-bit integer in the range [0 .. 2^32). If projectID is missing, then it is automatically assigned to 0

BTW, it actually called accountID, and also can have projectId. Maybe it is worth to rename it and add support for projectId?

Re "-1" as default for non-clustered version: user can omit this vmClusterTenantId option completely, and it will work as before, so upgrade would be safe with old configs, so I don't think it is worth to mention this option for single node vesrion/

Reply:
Sorry, I have not many experience with VM, so, please bear with me.
So, URL "http://127.0.0.1/select/0/prometheus/api/v1/query_range" will work even for single node VM, same as "http://127.0.0.1/api/v1/query_range" ?

Reply:
hi @deniszh 

> So, URL "http://127.0.0.1/select/0/prometheus/api/v1/query_range" will work even for single node VM

No.

Carbonapi detect VM version using url like http://127.0.0.1/metrics. This works both in single- and cluster-mode regardless of `vmClusterTenantID` option. But for query_range we have to use accountID in API calls for **cluster mode** only.

If user **do not** use vmClusterTenantID in carbonapi `config yaml`, then carbonapi will build single node VM API url: http://127.0.0.1/api/v1/query_range

If user **use** vmClusterTenantID in config yaml then (with this PR merged) cluster API will be used http://127.0.0.1/select/accountID/prometheus/api/v1/query_range for series.

Hope this helps.

BTW I changed code a bit so tenant ID can be configured more flexible, with both accoutID ant projectID.